changeset/spec/integration/changeset_spec.rb

```diff
@@ -160,6 +186,30 @@ class User < Dry::Struct
       expect(result.updated_at).to be_instance_of(Time)
     end
 
+    it 'preprocesses data using map blocks in a custom class' do
+      book = books.command(:create).call(title: 'rom-rb is awesome')
+
+      changeset_class = Class.new(ROM::Changeset::Update) do
+        map do |tuple|
+          extend_tuple(tuple)
+        end
+
+        private
+
+        def extend_tuple(tuple)
+          tuple.merge(title: tuple[:title] + ', yes really')
+        end
+      end
+
+      result = books.by_pk(book.id)
+        .changeset(changeset_class, title: 'rom-rb is awesome for real')
+        .commit
+
+      expect(result.id).to be(book.id)
+      expect(result.title).to eql('rom-rb is awesome for real, yes really')
+      expect(result.extended).to be_true
+    end
```

Comment:
This test fails, despite being almost identical to the test above (the only difference being it's an `Update` subclass, instead of `Create`).

If you put a `byebug` inside that map block, you'll see output like this:

```
(byebug) self
#<Class:0x00007fa9e6a39840>
```

Whereas if you put the byebug inside map block defined in the `Create` subclass in the test above, it shows:

```
(byebug) self
#<#<Class:0x00007ff5629fb8c0> relation=ROM::Relation::Name(books) data={:title=>"rom-rb is awesome"}>
```